relay-spans/src/span.rs

```diff
@@ -0,0 +1,459 @@
+    /// kept by the local machine where the span execution starts. On the server side, this
+    /// is the time when the server's application handler starts running.
+    /// Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+    ///
+    /// This field is semantically required and it is expected that end_time >= start_time.
+    pub start_time_unix_nano: i64,
+    /// end_time_unix_nano is the end time of the span. On the client side, this is the time
+    /// kept by the local machine where the span execution ends. On the server side, this
+    /// is the time when the server application handler stops running.
+    /// Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+    ///
+    /// This field is semantically required and it is expected that end_time >= start_time.
+    pub end_time_unix_nano: i64,
+    /// attributes is a collection of key/value pairs. Note, global attributes
+    /// like server name can be set using the resource API.
+    ///
+    /// The OpenTelemetry API specification further restricts the allowed value types:
+    /// <https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute>
+    /// Attribute keys MUST be unique (it is not allowed to have more than one
+    /// attribute with the same key).
+    #[serde(default)]
+    pub attributes: Vec<KeyValue>,
+    /// dropped_attributes_count is the number of attributes that were discarded. Attributes
+    /// can be discarded because their keys are too long or because there are too many
+    /// attributes. If this value is 0, then no attributes were dropped.
+    #[serde(default)]
+    pub dropped_attributes_count: u32,
+    /// events is a collection of Event items.``
+    #[serde(default)]
+    pub events: Vec<Event>,
+    /// dropped_events_count is the number of dropped events. If the value is 0, then no
+    /// events were dropped.
+    #[serde(default)]
+    pub dropped_events_count: u32,
+    /// links is a collection of Links, which are references from this span to a span
+    /// in the same or different trace.
+    #[serde(default)]
+    pub links: Vec<Link>,
+    /// links is a collection of Links, which are references from this span to a span
+    /// in the same or different trace.
+    #[serde(default)]
+    pub dropped_links_count: u32,
+    /// An optional final status for this span. Semantically when Status isn't set, it means
+    /// span's status code is unset, i.e. assume STATUS_CODE_UNSET (code = 0).
+    #[serde(default)]
+    pub status: Status,
+}
+
+impl OtelSpan {
+    /// sentry_status() returns a status as defined by Sentry based on the span status.
+    pub fn sentry_status(&self) -> &'static str {
+        let status_code = self.status.code.clone();
+
+        if status_code == StatusCode::Unset || status_code == StatusCode::Ok {
+            return "ok";
+        }
```

Comment:
Might be a matter or personal taste, but why not

```suggestion
        if self.status.code == StatusCode::Unset || self.status.code == StatusCode::Ok {
            return "ok";
        }
```

Reply:
I wanted to avoid the `clone` twice but since you added `Copy` above, I guess it's handled for me so sure.